src/core/drive/form_submission.ts

```diff
@@ -99,7 +99,9 @@ export class FormSubmission {
   }
 
   requestSucceededWithResponse(request: FetchRequest, response: FetchResponse) {
-    if (this.requestMustRedirect(request) && !response.redirected) {
+    if (response.unprocessableEntity) {
```

Comment:
Is this the only special case status we want to support?

Reply:
It might make sense to cover all `4xx` errors, not only `422`. 422 is pretty specific to rails scaffolding conventions where as `4xx` is more widely used for forms that have errors.

Reply:
`422` is problematic also for Rails. Default `HTML` response for invalid form is `200`, `422` is usually used only for `JSON` responses. I know it can be changed, but support for default behaviour would be nicer :).

Example from rails scaffold generator:
```ruby
  # POST /rooms
  # POST /rooms.json
  def create
    @room = Room.new(room_params)

    respond_to do |format|
      if @room.save
        format.html { redirect_to @room, notice: 'Room was successfully created.' }
        format.json { render :show, status: :created, location: @room }
      else
        format.html { render :new }
        format.json { render json: @room.errors, status: :unprocessable_entity }
      end
    end
  end
```